test/demo/eventListeners.inttest.ts

```diff
@@ -134,15 +137,25 @@ describe('Click listeners', () => {
                 });
 
                 test('The closeListener is called once with the correct metadata the other listeners are not called', async () => {
+                    expect(actionListener).toHaveBeenCalledTimes(1);
+                    expect(actionListener).toHaveBeenCalledWith({
+                        type: 'notification-action',
+                        notification: {...note, date: note.date.toJSON()},
+                        trigger: 'close',
+                        result: {task: 'close'}
+                    });
+
                     expect(closedListener).toHaveBeenCalledTimes(1);
                     expect(closedListener).toHaveBeenCalledWith({
                         type: 'notification-closed',
                         notification: {...note, date: note.date.toJSON()}
                     });
 
-                    // No 'action', or additional 'created', events
+                    // Action event fired before closed event
+                    expect(eventOrdering).toEqual(['created', 'action', 'closed']);
+
+                    // No additional 'created' events
                     expect(createdListener).toHaveBeenCalledTimes(1);
-                    expect(actionListener).toHaveBeenCalledTimes(0);
```

Comment:
Can we have a similar test (or tests?) for the different methods of closing? Like clicking the notification body?

Reply:
Can't fully do that before SERVICE-580, right? Can add a test that checks that removing the notification programmaticaly also triggers the event, but the others I think will need to wait

Reply:
Oh, we have no tests around `clear` and `clearAll`, will add

Reply:
Ahh yeah, true. We still have closing via `closeAll`, but suspect that uses the same flow within the provider?..

Will cover closing via select/button click with SERVICE-580.